llama-index-core/llama_index/core/workflow/context.py

```diff
@@ -117,25 +126,39 @@ def from_dict(
         workflow: "Workflow",
         data: Dict[str, Any],
         serializer: Optional[BaseSerializer] = None,
+        return_in_progress: bool = True,
     ) -> "Context":
         serializer = serializer or JsonSerializer()
 
         context = cls(workflow, stepwise=data["stepwise"])
         context._globals = context._deserialize_globals(data["globals"], serializer)
-        context._queues = {
-            k: context._deserialize_queue(v, serializer)
-            for k, v in data["queues"].items()
-        }
         context._streaming_queue = context._deserialize_queue(
             data["streaming_queue"], serializer
         )
         context._events_buffer = {
             k: [serializer.deserialize(ev) for ev in v]
             for k, v in data["events_buffer"].items()
         }
+        if len(context._events_buffer) == 0:
+            context._events_buffer = defaultdict(list)
         context._accepted_events = data["accepted_events"]
         context._broker_log = [serializer.deserialize(ev) for ev in data["broker_log"]]
         context.is_running = data["is_running"]
+        if return_in_progress:
```

Comment:
Can we add a comment about what this option does or what this code is doing? I'm actually a little confused (seems like the block conditions are inversed?)

Reply:
Sorry for the confusion...

I think we should remove the flag and just reload back whatever was in progress back in their respective queues.

Consider this situation:
- A workflow with steps: Step1 and Step2
- say at end of Step2 we create a checkpoint, but Step1 is actually in progress
- if we don't maintain what is currently in progress in the Context object, then the created checkpoint will forget that Step1 is currently in progress, and because its input event was already popped from the queue, it will be missed in Context.to_dict() at time of checkpoint creation

So, what this PR introduces is a new state key called `in_progress`, where active steps and their input events are being maintained. Once the step completes, this get's removed.